src/controller_examples/zookeeper_controller/exec/zookeeper_lib/helper_funcs.rs

```diff
@@ -9,62 +9,65 @@ use vstd::{prelude::*, string::*};
 
 verus! {
 
+pub struct ZKNodeResult {
+    pub res: Result<(), Error>,
+}
+
 struct LoggingWatcher;
 impl Watcher for LoggingWatcher {
     #[verifier(external)]
     fn handle(&self, e: WatchedEvent) {}
 }
 
 #[verifier(external)]
-pub fn reconcile_zk_node(zk: &ZookeeperCluster) -> Result<(), Error>
-    requires
-        zk@.metadata.name.is_Some(),
-        zk@.metadata.namespace.is_Some(),
-{
-    let path = cluster_size_zk_node_path(&zk);
+pub fn reconcile_zk_node(path: String, uri: String, replicas: String) -> ZKNodeResult {
     let zk_client = ZooKeeper::connect(
-        zk_service_uri(&zk).as_str().into_rust_str(),
+        uri.as_str().into_rust_str(),
         Duration::from_secs(10),
         LoggingWatcher,
     );
     if zk_client.is_err() {
-        return Result::Err(Error::ClusterSizeZKNodeCreationFailed);
+        return ZKNodeResult{ res: Err(Error::ClusterSizeZKNodeCreationFailed) };
     }
     let zk_client = zk_client.unwrap();
     let path_look_up = zk_client.exists(path.as_rust_string_ref(), false);
     if path_look_up.is_err() {
-        return Result::Err(Error::ClusterSizeZKNodeCreationFailed);
+        return ZKNodeResult{ res: Err(Error::ClusterSizeZKNodeCreationFailed) };
     }
     let path_look_up = path_look_up.unwrap();
     match path_look_up {
-        Some(_) => Ok(()),
+        Some(_) => {
+            return ZKNodeResult{ res: Ok(())};
```

Comment:
Does it mean that we will never update the `CLUSTER_SIZE` once the zk node gets created? Then how to support downscaling?

Reply:
@euclidgame @ZichengMa Can you address this one?

Reply:
I will later add the Update operation and get familar with how to import third party APIs.